src/Components/FollowButton/FollowArtistButton.tsx

```diff
@@ -113,24 +137,52 @@ export class FollowArtistButton extends React.Component<Props> {
     }
   }
 
-  render() {
-    const { artist, useDeprecatedButtonStyle, buttonProps, render } = this.props
+  closePopover() {
+    this.setState({ openSuggestions: false })
+  }
 
+  render() {
+    const {
+      artist,
+      useDeprecatedButtonStyle,
+      buttonProps,
+      render,
+      user,
+    } = this.props
+    const { openSuggestions } = this.state
     // FIXME: Unify design language
     const Button = useDeprecatedButtonStyle
       ? FollowButtonDeprecated
       : FollowButton
 
+    const popoverBody = [
+      <PopoverTitle onClose={() => this.closePopover()} />,
+      <SuggestionsPopover user={user} suggested={artist} />,
+    ]
     // Custom button renderer
     if (render) {
-      return <span onClick={this.handleFollow}>{render(artist)}</span>
+      return (
+        <Popover
+          onOuterAction={() => this.closePopover()}
+          isOpen={openSuggestions}
+          body={popoverBody}
+        >
+          <span onClick={this.handleFollow}>{render(artist)}</span>
+        </Popover>
+      )
     } else {
       return (
-        <Button
-          isFollowed={artist && artist.is_followed}
-          handleFollow={this.handleFollow}
-          buttonProps={buttonProps}
-        />
+        <Popover
+          onOuterAction={() => this.closePopover()}
+          isOpen={openSuggestions}
+          body={popoverBody}
+        >
+          <Button
+            isFollowed={artist && artist.is_followed}
+            handleFollow={this.handleFollow}
+            buttonProps={buttonProps}
+          />
+        </Popover>
```

Comment:
I actually don't think that's the best division b/c that will require places that want to use an artist follow button with popover suggestions go thru that dance. Ideally (like on current artsy.net pages), we do want the artist suggestions to always show up as a popover once you click 'Follow'.

I added the prop here `triggerSuggestions` to be able to control how thats rolled out (since this isnt styled/finished yet, we dont want to just start seeing it everywhere). But then eventually yea, we wouldn't _just_ want it on the follow action in the artwork page sidebar, we'd want it in the artist header, etc. - everywhere.

So I think it's actually fine for it to exist here?

Reply:
It seems like there's a lot going on in that button, and given the fact that there are custom renderers used all over the place now, including in the follow partner button, etc; everything inside that button is visually deprecated and that button is already overly complex.

Feel free to do whatever, but basically how I imagine it is the Popover contains a child and if you click the child (the follow artist button) it triggers a mutation and re-calls the render function up above with `is_following`. That should then say to Popover `show={true}`, which is visually centered on the containing "child" element. And if you want to share in various places in force, just use composition; export a wrapper component from the file that contains the pattern described above, that way things can stay decoupled. Soon there will be a ton of positioning logic inside of FollowArtistButton to handle different display cases when really all of that should be contained inside of Popover. 

```js

<FollowArtistButton 
  render={({ is_following }) => {
    return (
      <Popover show={is_following}>
          <Button size='large'>{is_following ? Unfollow : Follow}</Button>
      </Popover>
    )
  }}
/>
```